src/spatialdata_plot/pl/utils.py

```diff
@@ -808,44 +800,43 @@ def _map_color_seg(
     seg_boundaries: bool = False,
 ) -> ArrayLike:
     cell_id = np.array(cell_id)
-    if color_vector is not None and isinstance(color_vector.dtype, pd.CategoricalDtype):
-        # users wants to plot a categorical column
+
+    if pd.api.types.is_categorical_dtype(color_vector.dtype):
+        # Case A: users wants to plot a categorical column
         if np.any(color_source_vector.isna()):
             cell_id[color_source_vector.isna()] = 0
-        val_im: ArrayLike = map_array(seg, cell_id, color_vector.codes + 1)
+        val_im: ArrayLike = map_array(seg.copy(), cell_id, color_vector.codes + 1)
```

Comment:
hmm is the copy really required here? I thought that map array returns an array with same shape as the original array, but it would not have the same pointer

Reply:
Ah, I remember. Don't ask me why, but if I don't `copy()`, one of the tests fails with:

`ValueError: buffer source array is read-only`

Reply:
I had the same issue with the current `main`: whenever I tried to color labels by a column in `table.obs`, I'd get `ValueError: buffer source array is read-only`, and putting in those `.copy()` calls fixed it. This happened regardless of whether the `SpatialData` object was buffered to disk or not.

Reply:
I see, this is probably a bug in skimage actually. For now we could keep it as it is, but please add a comment with the issue so we remove this when we can.